lib/get-category-path.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var path = require('path'),
+  utility = require('./utility');
+
+
+/**
+ * Determines the category directory structure from the pattern's data
+ *
+ * @param {Object} patternObject  individual pattern's data (from /pattern-name/pattern.yml)
+ * @param {Object} options  pattern-importer options
+ *
+ * @return {String} categoryPath  internal path url to category
+ *
+ * @requires ./lib/utility
+ * @requires ./lib/category-name-converter
+ */
+exports.getCategoryPath = function (patternObject, options) {
```

Comment:
@scottnath I think the first comment in this block by @ericduran refers to Drupal's coding standards, which I proposed, but would unfortunately entail a major overhaul of all your code, most of which doesn't affect functionality.

In regard to Eric's second point, we do seem to have an awful lot of code for an awful lot of very simple utilities. I also know I don't know too much about the higher-level structure of things here. But the sooner I know, the sooner I can make recommendations on how not to create a black hole singularity. I think actual suggestions for improvement will come later as I get to know what we're trying to do more.